.github/workflows/release.yaml

```diff
@@ -0,0 +1,52 @@
+name: Weekly Release
+
+on:
+  schedule:
+    - cron: '0 11 * * 1' # Run every Monday at 7am NYC time
+      branches:
+        - master
+  workflow_dispatch: # allows manual releasing
+      branches:
+        - master
+jobs:
+  build_and_release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Bazel
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'adopt'
+          java-version: '21'
+
+      - name: Build with Bazel
+        run: bazel build java/com/google/copybara/copybara_deploy.jar
+
+      - name: Get current date
+        id: date
+        run: echo "::set-output name=date::$(date +%Y%m%d)"
+
+      - name: Create Release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: v${{ steps.date.outputs.date }}
+          release_name: Release v${{ steps.date.outputs.date }}
+          body: |
+            Automated weekly test release snapshot from master branch.
+            This is a test release, version compatibility or correctness
```

Comment:
what about publishing binaries for non-test releases?

You might want to look at how we do this for Bazel ruleset releases. There is one workflow for "a tag was created"
https://github.com/bazel-contrib/rules-template/blob/main/.github/workflows/release.yml
which covers tags applied by engineers when we want to release.

Then there's a separate cron-scheduled "make sure we release sometimes" https://github.com/bazel-contrib/rules-template/blob/main/.github/workflows/tag.yaml which triggers the other one.

Reply:
For now "test" releases is what we can commit to. Hopefully over time we can have more supported releases.

Thanks for the pointers! Some folks in Google also gave feedback on the workflow.